### PR TITLE
Fail on startup for Http/2 HTTPS on Mac

### PR DESCRIPTION
**Pull request by Tratcher**
#11061 Testing via Helix

### CODE REVIEW

**Comment by Tratcher** [comment]
It worked, and failed a bunch of other tests like Templates.Test.GrpcTemplateTest.GrpcTemplate 😁.

File: src/ProjectTemplates/test/GrpcTemplateTest.cs [hunk 1]
```diff
@@ -35,6 +36,13 @@ public async Task GrpcTemplate()
             var buildResult = await Project.RunDotNetBuildAsync();
             Assert.True(0 == buildResult.ExitCode, ErrorMessages.GetFailedProcessMessage("build", Project, buildResult));
 
+            // These templates are HTTPS + HTTP/2 only which is not supported on Mac due to missing ALPN support.
```
Comment (ryanbrandenburg) [inline src/ProjectTemplates/test/GrpcTemplateTest.cs]: Shouldn't this be:
```c#
[ConditionalFact]
[OSSkipCondition(OperatingSystems.MacOSX)]
```
Instead? This pattern gives the impression that the test ran on Mac when it didn't do anything interesting. Either that or we should verify the behavior you get on a mac (I assume it gives a semi-useful error message?).

File: src/Servers/Kestrel/Kestrel/test/KestrelConfigurationBuilderTests.cs [hunk 2]
```diff
@@ -371,11 +384,23 @@ public void DefaultConfigSectionCanSetProtocols(string input, HttpProtocols expe
             Assert.True(ran3);
         }
 
+        [Theory]
+        [InlineData("http1", HttpProtocols.Http1)]
+        // [InlineData("http2", HttpProtocols.Http2)] // Not supported due to missing ALPN support. https://github.com/dotnet/corefx/issues/33016
```
Comment (ryanbrandenburg) [inline src/Servers/Kestrel/Kestrel/test/KestrelConfigurationBuilderTests.cs]: Too bad you can't mark InlineData as skipped.

**Review by ryanbrandenburg — APPROVED** [review]
This is good to go once the ProjectTemplate test either uses OSSkipCondition or checks the behavior that was changed here.

File: src/ProjectTemplates/test/GrpcTemplateTest.cs (hunk 1 above)
Comment (Tratcher) [reply src/ProjectTemplates/test/GrpcTemplateTest.cs]: Added a check for the error message.

**Pull request merged**